### PR TITLE
feat: dropdown huge lend amount enabled for logged in users

### PR DESCRIPTION
**Pull request by roger-in-kiva**
- dropdown huge lend amount enabled for logged in users under experiment flag

### CODE REVIEW

File: src/components/LoanCards/ListLoanCard.vue [hunk 1]
```diff
@@ -133,6 +135,8 @@
 						:amount-left="amountLeft"
 						:show-now="!enableFiveDollarsNotes"
 						:enable-five-dollars-notes="enableFiveDollarsNotes"
+						:enable-huge-amount="enableHugeAmount"
+						:is-visitor="isVisitor"
```
Comment (dyersituations) [inline src/components/LoanCards/ListLoanCard.vue:139]: Why was `isVisitor` added to these two instances of action button?

File: src/components/LoanCards/GridLoanCard.vue [hunk 2]
```diff
@@ -49,6 +49,8 @@
 					:amount-left="amountLeft"
 					:show-now="!enableFiveDollarsNotes"
 					:enable-five-dollars-notes="enableFiveDollarsNotes"
+					:enable-huge-amount="enableHugeAmount"
+					:is-visitor="isVisitor"
```
Comment (dyersituations) [inline src/components/LoanCards/GridLoanCard.vue:53]: Same question here, why was `isVisitor` added here?

File: src/pages/Checkout/CheckoutPage.vue [hunk 3]
```diff
@@ -585,6 +589,9 @@ export default {
 
 		this.initializeFiveDollarsNotes();
 
+		// Enable huge lend amount
+		this.initializeHugeLendAmount();
```
[the remark this answers is not in the document]
Comment (roger-in-kiva) [reply src/pages/Checkout/CheckoutPage.vue:593]: Following the `initializeFiveDollarsNotes` test pattern to this case to have the flag before mounting child components. Make sense?

File: src/components/Checkout/LoanPrice.vue [hunk 4]
```diff
@@ -114,9 +122,10 @@ export default {
 
 				const minAmount = parseFloat(this.minAmount || 25); // 25_hard_coded
 
-				const priceArray = getDropdownPriceArrayCheckout(remainingAmount, minAmount, this.enableFiveDollarsNotes); // eslint-disable-line max-len
-				if (!priceArray.includes(Number(this.price).toFixed())) {
-					priceArray.push(Number(this.price).toFixed());
+				const showHugeAmount = this.enableHugeAmount && this.isLoggedIn;
+				const priceArray = getDropdownPriceArrayCheckout(remainingAmount, minAmount, this.enableFiveDollarsNotes, showHugeAmount); // eslint-disable-line max-len
+				if (!priceArray.includes(numeral(this.price).format('0,0'))) {
```
Comment (dyersituations) [inline src/components/Checkout/LoanPrice.vue:127]: Here you changed the formatting. Why was that needed?

File: src/components/LoanCards/GridLoanCard.vue (hunk 2 above)
Comment (roger-in-kiva) [reply src/components/LoanCards/GridLoanCard.vue:53]: In order to generate the huge amount price array for both conditions, logged in users and experiment flag.

File: src/util/loanUtils.js [hunk 5]
```diff
@@ -93,21 +93,60 @@ export function build5DollarsPriceArray(amountLeft) {
 	return priceArray;
 }
 
-export function getDropdownPriceArray(unreservedAmount, minAmount, enableFiveDollarsNotes, inPfp = false) {
+function buildHugePriceArray(amountLeft) {
```
Comment (dyersituations) [inline src/util/loanUtils.js:96]: Since these utils are critical in our app, new unit tests should be added for these changes.

File: src/components/Checkout/LoanPrice.vue (hunk 4 above)
Comment (roger-in-kiva) [reply src/components/Checkout/LoanPrice.vue:127]: Mainly for the case of $1,000 coming from shared loan cards. Not affecting other cases.

File: src/util/loanUtils.js (hunk 5 above)
Comment (roger-in-kiva) [reply src/util/loanUtils.js:96]: Agreed! I'm on it.

File: src/components/Checkout/LoanPrice.vue (hunk 4 above)
Comment (dyersituations) [reply src/components/Checkout/LoanPrice.vue:127]: Does it affect the existing formatting? And the comparison still works, since the formatting used to be different?

**Comment by roger-in-kiva** [comment]
> Looks like existing unit tests are breaking.

Yep, working on it.

File: src/components/Checkout/LoanPrice.vue (hunk 4 above)
Comment (roger-in-kiva) [reply src/components/Checkout/LoanPrice.vue:127]: I tested different situations and comparison still works for current and new cases.

File: src/util/loanUtils.js (hunk 5 above)
Comment (roger-in-kiva) [reply src/util/loanUtils.js:96]: Added!

File: src/util/loanUtils.js [hunk 6]
```diff
@@ -93,21 +93,60 @@ export function build5DollarsPriceArray(amountLeft) {
...
+	const minAmount = 100;
+	const limitAmount = amountLeft > 1000 ? 1000 : amountLeft;
+	const N = limitAmount / minAmount;
```
Comment (dyersituations) [inline src/util/loanUtils.js:99]: `N` should probably be renamed to something more consistent with other variable names.

Comment (roger-in-kiva) [reply src/util/loanUtils.js:99]: Agreed, but should apply for all cases we have this.
